### PR TITLE
Add complete daily ecdr

### PR DESCRIPTION
**Pull request by sc0tts**
Still a few items remaining, eg making sure pass1, pass2 subdirs are properly passed throughout the complete_daily processing stream.

Also need to implement eCDR-specific file name templates.

And some of the QA fields may not be getting properly set yet.

But this will yield initial complete daily fields so that further work -- eg start of monthly composite files -- and comparisons with cdrv4 files can begin

### CODE REVIEW

File: seaice_ecdr/complete_daily_ecdr.py [hunk 1]
```diff
@@ -0,0 +1,545 @@
...
+    if hemisphere == NORTH and resolution == "12":
+        array_shape = (896, 608)
+    elif hemisphere == SOUTH and resolution == "12":
+        array_shape = (664, 632)
+    else:
+        raise RuntimeError(
+            f"Could not determine array shape for {hemisphere}" f" and {resolution}"
+        )
+    array = np.zeros(array_shape, dtype=dtype)
+    array[:] = fill_value
```
Comment (trey-stafford) [inline seaice_ecdr/complete_daily_ecdr.py:148]: These last two lines could be reduced to one:

```
array = np.full(array_shape, fill_value)
```

https://numpy.org/doc/stable/reference/generated/numpy.full.html

File: seaice_ecdr/complete_daily_ecdr.py [hunk 2]
```diff
@@ -0,0 +1,545 @@
...
+    required=True,
+    type=click.Path(
+        exists=True,
+        file_okay=False,
+        dir_okay=True,
+        writable=True,
+        resolve_path=True,
+        path_type=Path,
+    ),
+    default=INITIAL_DAILY_OUTPUT_DIR,
```
Comment (trey-stafford) [inline seaice_ecdr/complete_daily_ecdr.py:499]: Create seprate dir for complete daily files

File: seaice_ecdr/complete_daily_ecdr.py [hunk 3]
```diff
@@ -0,0 +1,545 @@
...
+    create_cdecdr_for_date_range(
+        hemisphere=hemisphere,
+        start_date=date,
+        end_date=date,
+        resolution=resolution,
+        output_dir=output_dir,
+    )
+
+
+if __name__ == "__main__":
```
Comment (trey-stafford) [inline seaice_ecdr/complete_daily_ecdr.py:533]: clean this up before merge

**Review by trey-stafford — CHANGES_REQUESTED** [review]
Great start! Just a few comments to address.

File: seaice_ecdr/complete_daily_ecdr.py (hunk 2 above)
Comment (sc0tts) [reply seaice_ecdr/complete_daily_ecdr.py:499]: I added ide_dir, tie_dir, and output_dir in most function calls that need or pass-through dir names

File: seaice_ecdr/complete_daily_ecdr.py (hunk 3 above)
Comment (sc0tts) [reply seaice_ecdr/complete_daily_ecdr.py:533]: By "clean up", do you mean remove the code that calls the make_cdecdr_netcdf() routine if the file is executed?

File: seaice_ecdr/complete_daily_ecdr.py [hunk 4]
```diff
@@ -0,0 +1,545 @@
...
+    )
+    ecdr_filepath = Path(
+        directory_name,
+        ecdr_filename,
+    )
+
+    return ecdr_filepath
+
+
+# TODO: Perhaps this isn't needed?  The necessary precursor files can
```
[the remark this answers is not in the document]
Comment (sc0tts) [reply seaice_ecdr/complete_daily_ecdr.py:85]: Removed.  This date iterator is not needed with the recursive code we're using.

File: seaice_ecdr/complete_daily_ecdr.py (hunk 3 above)
Comment (trey-stafford) [reply seaice_ecdr/complete_daily_ecdr.py]: Yes, I cleaned this up and added the complete CLI to the standard entrypoint. E.g., this is now possible:

```
$ ./scripts/cli.sh  cdecdr --date 2022-03-02 --hemisphere north --resolution 12
```

**Pull request merged**